### ISSUE TITLE
Claiming / provisioning a devices without an deviceID

### ISSUE THREAD

**Issue by ojna** [Brewskey/spark-server#296]
Looking at the official particle api docs you can see that the endpoint [POST /v1/devices](https://docs.particle.io/reference/device-cloud/api/#provision-a-new-device-id) should be able to both claim a device with id or provision a new device if it's left out and supplied with a product id. 

This applies to eg Raspberry Pi. I've been messing around with the code in the dev branch and everything works fine with the Particle Agent för RPI if the endpoints creates an ID and then responds with:

```json
{ "ok": true, "device_id": "new-device-id" }
```

Once its stored in the database if works fine with the rest of the cli and so on. But i guess we need to make sure the generated id don't collide with other ids later on.

Are there any roadmap to adressing this, or is anyone working on it? Either case i will fix this and with some coordination with this project i can send a pull request with it later on.

**Comment by jlkalberer** [comment]
Please send a pull request.

I don't completely understand what the problem is from your description ("the generated id don't collide with other ids later on") but I'm not doing any work on this repro unless it's a bug fix.

**Comment by ojna** [comment]
I will, but are there any information on my question?

> Are there any roadmap to adressing this, or is anyone working on it?

**Comment by jlkalberer** [comment]
No, I don't have plans on fixing this as it isn't a critical bug

**Comment by jlkalberer** [comment]
Hey @ojna

I started looking into implementing this but the API is behind the enterprise plan (as far as I can tell). I don't want to pay for it or start a trial.

**Comment by ojna** [comment]
Hi,

I've done some work on this and will continue this week. Will post a pull request when its ready for some feedback.
